python/cudf/cudf/core/udf/typing.py

```diff
@@ -37,9 +37,18 @@ class MaskedType(types.Type):
     def __init__(self, value):
         # MaskedType in Numba shall be parameterized
         # with a value type
-        if not isinstance(value, SUPPORTED_NUMBA_TYPES):
-            raise TypeError("value_type must be a numeric scalar type")
-        self.value_type = value
+        if isinstance(value, SUPPORTED_NUMBA_TYPES):
+            self.value_type = value
+        else:
+            # Unsupported Dtype. Numba tends to print out the type info
+            # for whatever operands and operation failed to type and then
+            # output its own error message. Putting the message in the repr
+            # then is one way of getting the true cause to the user
+            self.value_type = types.Poison(
+                "\n\n\n Unsupported MaskedType. If you are seeing this, "
+                "you are likely trying to use a column of unsupported "
+                f"dtype in a UDF. Unsupported type: {value} \n\n\n"
```

Comment:
Could we avoid using second person? Also is it _likely_ the cause, or certainly the cause? If that is the only path to this error, I would remove "likely." (Sorry for having suggestions in the same place twice, I didn't notice the first time.)
```suggestion
                "\n\n\nUnsupported MaskedType. This is likely caused by "
                "using a column of unsupported dtype in a UDF. Unsupported "
                f"type: {value}\n\n\n"```

Reply:
The intent is that functionally this only shows up in this exact case. The problem is that the way Numbas typing pass works, it seems like one could conceivably see this print under certain conditions while debugging, or perhaps unintentionally if cuDF errors somewhere else for an unrelated reason and prints out the values of some variables in the traceback.